Fixes #20524: Enhance API script scheduling validation

pheus opened the pull request:
### Fixes: #20524

Tightens script scheduling validation:

- Rejects `schedule_at` values in the past.
- When only `interval` is provided, starts now and repeats.
- Uses `local_now()` for accurate, timezone‑aware comparisons.

Small, backward‑compatible fix to make scheduling behave as expected.

jnovinger reviewed (commented):
Thanks @pheus. This fix looks good, although I did note one other edge case (w/ a suggestion) that the original bug report did not mention. Can you please address that, since we're already making changes there?

Could you please also add the following tests for this? Probably in `ScriptTest` around line 910 in `netbox/extras/tests/test_api.py`:

```python
def test_schedule_script_past_time_rejected(self):
    """Scheduling with past schedule_at should 
fail"""
    # POST with schedule_at = timezone.now() - timedelta(hours=1)
    # Expect 400 with "must be in the future" error

def test_schedule_script_interval_only(self):
    """Interval without schedule_at should auto-set 
schedule_at to now"""
    # POST with interval=60, no schedule_at
    # Verify job created with schedule_at set

def test_schedule_script_when_disabled(self):
    """Scheduling should fail when 
script.scheduling_enabled=False"""
    # POST with schedule_at or interval
```

Let me know if you have any questions.

pheus committed:
feat(extras): Enhance API script scheduling validation

Add validation for script scheduling to ensure the `schedule_at` time
is in the future and default to the current time when only `interval`
is provided. Incorporate `local_now` for accurate time comparisons.

Fixes #20524

pheus commented on netbox/extras/tests/test_api.py:
```diff
@@ -878,14 +879,16 @@ def run(self, data, commit=True):
 
+    @property
     def python_class(self):
         return self.TestScriptClass
```
Make this a property to mirror the model’s `cached_property`. This returns the class object (not a bound function), which fixes the test setup and matches production behavior.

pheus commented:
Thanks for the thoughtful review!

I’ve pushed updates to address your suggestion and added tests:

- Use `.get()` so `{"schedule_at": null, "interval": 60}` starts now (no start‑less recurring job).
- Tests added: past `schedule_at` → 400; `interval`‑only → auto‑sets `schedule_at`; scheduling disabled → reject.
- Fixed the test setup to return the class from `python_class`, matching production behavior.

Happy to tweak anything.
